iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    TokenCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">SAS <see cref="AzureSasCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    AzureSasCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+```
```

Comment:
Q - What about the "ConnectionStringCredential" factory method (the one that'll let people specify the ttl for their sas tokens)? Is that a stretch goal?

Reply:
That should be user defined in their implementation. If they have a one time ttl, the token will only be valid for a while. If they have implementation to renew the token, our SDK will automatically pick up the updated token. Apart from that, our existing ConnectionString input support continues to exist.